### PR TITLE
Add arm models resources test case

### LINKED ISSUE #2648: Need to add arm models resources test case

### PR DESCRIPTION
**Pull request by v-jiaodi**
fixes #2648

### CODE REVIEW

File: packages/typespec-ts/src/modular/buildSerializeUtils.ts [hunk 1]
```diff
@@ -198,7 +198,8 @@ export function isSpecialUnionVariant(
         })
         ?.some((p) => {
           return isSpecialUnionVariant(p, [...variantStack, p]);
-        }))
+        })) ||
+    (t.type === "enum" && !(t.isFixed || t.isNonExhaustive))
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildSerializeUtils.ts:202]: if union has a variant of enum, it needs special casting.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 2]
```diff
@@ -658,8 +658,15 @@ function buildBodyParameter(
...
+      let bodyElementStatement = "";
+      if (
+        isTypeNullable(bodyParameter.type.elementType) ||
+        bodyParameter.type.elementType.optional
+      ) {
+        bodyElementStatement = `!p ? p : `;
+      }
       return `\nbody: (${bodyParameter.clientName} ?? []).map((p) => { 
-      return ${mapBody};
+      return ${bodyElementStatement}${mapBody};
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:669]: to handle array elementType could be nullable or undefined.

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 3]
```diff
@@ -1714,7 +1714,10 @@ function emitCredentialParam(
         credential_types.push(type);
       }
     }
-    if (credential_types.length > 0) {
+    if (
+      credential_types.length > 0 &&
+      context.rlcOptions?.addCredentials !== false
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1719]: a bug fix for classical layer and api layer doesn't respect the addCredentials: false configuration.

File: packages/typespec-ts/test/integration/generated/client/naming/src/clientDefinitions.ts [hunk 4]
```diff
@@ -39,19 +39,19 @@ export interface Parameter {
 
 export interface PropertyClient {
   post(
-    options?: PropertyClientParameters,
+    options: PropertyClientParameters,
```
Comment (qiaozha) [inline packages/typespec-ts/test/integration/generated/client/naming/src/clientDefinitions.ts:42]: https://github.com/Azure/cadl-ranch/commit/fd61404ddd3d374be328af588b11985af4869189#diff-84598cc2d97c5c2a7f950a9c4d97316840c05c6f611030c2f97d812e9480b345 it should be expected.

**Review by qiaozha — APPROVED** [review]

File: packages/typespec-ts/test/integration/generated/azure/resource-manager/models/resources/samples-dev/nestedProxyResourcesCreateOrReplaceSample.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+dotenv.config();
+
+/**
+ * This sample demonstrates how to call operation CreateOrReplace
+ *
+ * @summary call operation CreateOrReplace
+ */
+async function nestedProxyResourcesCreateOrReplaceSample() {
+  const client = createAzureArmResourceClient();
```
Comment (lirenhe) [inline packages/typespec-ts/test/integration/generated/azure/resource-manager/models/resources/samples-dev/nestedProxyResourcesCreateOrReplaceSample.ts:17]: Shall we exclude RLC samples for resource manager SDKs?

Comment (lirenhe) [reply packages/typespec-ts/test/integration/generated/azure/resource-manager/models/resources/samples-dev/nestedProxyResourcesCreateOrReplaceSample.ts:17]: seems we are generating RLC for mgmt. SDK,  is this needed?

Comment (qiaozha) [reply packages/typespec-ts/test/integration/generated/azure/resource-manager/models/resources/samples-dev/nestedProxyResourcesCreateOrReplaceSample.ts:17]: We don't object to generate standalone RLC for mgmt plane and if we split RLC from Modular, both will be a requirement for mgmt plane, though modular is the current recommendation here. Another consideration is this is for test coverage.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**